scripts/tests/twister/test_config_parser.py

```diff
@@ -0,0 +1,265 @@
+    value = "10"
+    typestr = "int"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == 10
+
+
+def test_cast_string_to_bool(zephyr_base):
+    value = True
+    typestr = "bool"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result is True
+
+
+def test_cast_string_to_map(zephyr_base):
+    value = "map1 map2"
+    typestr = "map"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+    result = parser._cast_value(value, typestr)
+
+    assert result == "map1 map2"
+
+
+def test_raises_error_invalid_cast(zephyr_base):
+    value = "test"
+    typestr = "int"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+
+    with pytest.raises(ValueError):
+        parser._cast_value(value, typestr)
+
+
+def test_cast_string_to_unknown_type(zephyr_base):
+    value = "test"
+    typestr = "unknown"
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    parser = TwisterConfigParser("config.yaml", loaded_schema)
+
+    with pytest.raises(ConfigurationError):
+        parser._cast_value(value, typestr)
+
+
+def test_load_invalid_test_config_yaml(zephyr_base):
+    filename = "test_data.yaml"
+
+    yaml_data = '''
+    gibberish data
+    '''
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas','twister', 'test-config-schema.yaml')
+    )
+
+    with mock.patch('builtins.open', mock.mock_open(read_data=yaml_data)):
+        parser = TwisterConfigParser(filename, loaded_schema)
+        with pytest.raises(Exception):
+            parser.load()
+
+
+def test_load_yaml_with_no_scenario_data(zephyr_base):
+    filename = "test_data.yaml"
+
+    yaml_data = '''
+    tests:
+      common:
+        extra_args: '--CONF_FILE=file2.conf --OVERLAY_CONFIG=config2.conf'
+    '''
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas','twister', 'testsuite-schema.yaml')
+    )
+
+    with mock.patch('builtins.open', mock.mock_open(read_data=yaml_data)):
+        parser = TwisterConfigParser(filename, loaded_schema)
+        parser.load()
+
+    with pytest.raises(KeyError):
+        scenario_data = parser.get_scenario('scenario1')
+        assert scenario_data is None
```

Comment:
Seems similar. Investigate parametrisation possibilities.

Reply:
You are probably right - it is a good candidate for refactoring, but I guess it may impact code readability.